lib/src/skin_tone_overlay.dart

```diff
@@ -0,0 +1,111 @@
+import 'package:emoji_picker_flutter/emoji_picker_flutter.dart';
+import 'package:flutter/material.dart';
+
+/// Overlay for SkinTone
+OverlayEntry buildSkinToneOverlay(
+  BuildContext context,
+  Emoji emoji,
+  double emojiSize,
+  CategoryEmoji? categoryEmoji,
+  int index,
+  int skinToneCount,
+  Config config,
+  double scrollControllerOffset,
+  double tabBarHeight,
+  EmojiPickerUtils utils,
+  OnEmojiSelected onEmojiSelected,
+) {
+  // Generate other skintone options
+  final skinTonesEmoji = SkinTone.values
+      .map((skinTone) => utils.applySkinTone(emoji, skinTone))
+      .toList();
+
+  final position = _calculateEmojiPosition(context, index, config.columns,
+      skinToneCount, scrollControllerOffset, tabBarHeight);
+
+  return OverlayEntry(
+    builder: (context) => Positioned(
+      left: position.left,
+      top: position.top,
+      child: Material(
+        elevation: 4.0,
+        child: Container(
+          padding: const EdgeInsets.symmetric(vertical: 4.0),
+          color: config.skinToneDialogBgColor,
+          child: Row(
+            children: [
+              SizedBox(
+                width: emojiSize,
+                height: emojiSize,
+                child: EmojiCell.fromConfig(
+                  categoryEmoji: categoryEmoji,
+                  config: config,
+                  emoji: emoji,
+                  emojiSize: emojiSize,
+                  onEmojiSelected: onEmojiSelected,
+                ),
+              ),
+              EmojiCell.fromConfig(
+                emoji: emoji,
+                emojiSize: emojiSize,
+                categoryEmoji: categoryEmoji,
+                onEmojiSelected: onEmojiSelected,
+                config: config,
+              ),
+              ...List.generate(
+                SkinTone.values.length,
+                (index) => EmojiCell.fromConfig(
+                  emoji: skinTonesEmoji[index],
+                  emojiSize: emojiSize,
+                  categoryEmoji: categoryEmoji,
+                  onEmojiSelected: onEmojiSelected,
+                  config: config,
+                ),
```

Comment:
I considered that when I wrote it, but ended up not doing it due to visibility. Now I'm fine with keeping it, thanks !